### PR TITLE
Add date observation value type

### PR DESCRIPTION
**Pull request by forus**

### CODE REVIEW

**Comment by forus** [comment]
TODO Add a test case and database migration script.

**Comment by ewelinagr** [comment]
Tested on Oracle, the script worked.

**Comment by forus** [comment]
Created TMT-418 to deal with date aggregates differently.

**Comment by forus** [comment]
TMT-420 is created to further improve on how do we deal with the date data type in constraints.

File: transmart-core-db/grails-app/domain/org/transmartproject/db/i2b2data/ObservationFact.groovy [hunk 1]
```diff
@@ -32,6 +32,9 @@ class ObservationFact implements Serializable {
     public static String TYPE_DATE = 'D'
     public static String TYPE_RAW_TEXT = 'B'
     public static String EMPTY_MODIFIER_CODE = '@'
+    public static Set<String> ALL_TYPES = [TYPE_TEXT, TYPE_NUMBER, TYPE_RAW_TEXT, TYPE_NUMBER]
```
Comment (ewelinagr) [inline transmart-core-db/grails-app/domain/org/transmartproject/db/i2b2data/ObservationFact.groovy]: TYPE_NUMBER duplicated. Did you mean TYPE_DATE?

Comment (forus) [reply transmart-core-db/grails-app/domain/org/transmartproject/db/i2b2data/ObservationFact.groovy]: Ah, good catch. It has to be `TYPE_DATE` of course. Thanks!

**Comment by ewelinagr** [comment]
Looks good! Only the conflict to be resolved. Are the e2e tests still passing?

File: transmart-data/updatedb/release-17.1-HYVE-5/postgres/to_date_data_type.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+BEGIN;
+DO LANGUAGE 'plpgsql' $$
+DECLARE
+cd_query varchar;
+BEGIN
+ cd_query := (select (string_agg('select ' || c_facttablecolumn || ' as concept_cd from ' || c_tablename || ' where ' || c_columnname || ' ' || c_operator || ' ''' || c_dimcode || ''' escape ''^'' ', ' union ') || ';') as query from i2b2metadata.i2b2_secure where c_visualattributes like '__D');
+ set search_path TO "$user", i2b2demodata;
+ EXECUTE format('CREATE TEMP TABLE tmp_concept_codes ON COMMIT DROP AS %s', cd_query);
```
Comment (gijskant) [inline transmart-data/updatedb/release-17.1-HYVE-5/postgres/to_date_data_type.sql]: @forus Fails if there is no such concept in the database (e.g., on transmart.thehyve.net)

**Pull request merged**